### PR TITLE
create is_logic_running method

### PR DESCRIPTION
**Pull request by proto3**
I'm in a case where I need to know if Logic was running before I try to launch it.
If Logic was running, my program will let it run when it terminate. Otherwise it would kill Logic after use to let the system in the same state as found.

### CODE REVIEW

File: saleae/saleae.py [hunk 1]
```diff
@@ -151,6 +152,16 @@ def kill_logic(kill_all=False):
 		for candidate in candidates:
 			candidate.terminate()
 
+	@staticmethod
+	def is_logic_running():
```
Comment (ppannuto) [inline saleae/saleae.py:169]: Could you add a short doc comment for this function here so this gets picked up by the automated tooling?

File: saleae/saleae.py [hunk 2]
```diff
@@ -136,6 +136,7 @@ def kill_logic(kill_all=False):
 		# conservative assumption that if there's only one process running with
 		# 'logic' in the name, that it's Saleae Logic. If kill_all is set to
 		# True, all processes with 'logic' in the name are killed.
+		# Note that is_logic_running method rely on the same trick.
```
Comment (ppannuto) [inline saleae/saleae.py]: Since there's already one other change, if you wouldn't mind fixing a small grammar nit too: s/rely/relies

**Review by ppannuto — DISMISSED** [review]
Small doc fixes, then looks good to me -- thanks!

**Comment by proto3** [comment]
I finally added a **_list_logic_candidates** function to factorize code between **kill_logic** and **is_logic_running**.
Currently the way how Logic software is found does not work for me, the Logic process is called '_Main_' on my system. So I assume **_list_logic_candidates** will change again and having it in one place is cleaner.

**Comment by ppannuto** [comment]
Thanks for this! Sorry I lost track of it and let it sit for a while :/

**Pull request merged**